File: packages/platform/src/ipc/setting.ts
/**
 * 用户设置 IPC 封装 — 用户级键值配置存取。
 *
 * 后端：壳层 `commands/setting.rs`（InMemoryUserSettingStore + app::setting::UserSettingService）。
 *
 * @author agent
 * @created 2026-08-13
 */

import { call } from "./invoke";

const OWNER_ID = 1; // 桌面单用户；多用户时由登录态注入

/** 读取单键设置。 */
export function userSettingGet(key: string): Promise<string | null> {
  return call<string | null>("user_setting_get", {
    request: { owner_id: OWNER_ID, key },
  });
}

/** 写入单键设置（空值删除）。 */
export function userSettingSet(key: string, value: string): Promise<void> {
  return call<void>("user_setting_set", {
    request: { owner_id: OWNER_ID, key, value },
  });
}

File: apps/desktop/src-tauri/src/platforms/xianyu/ipc/setting.rs
//! 用户设置 Tauri commands — 用户级键值存储。

use crate::platforms::xianyu::persist::InMemoryUserSettingStore;
use crate::shared::ipc::IpcResponse;
use common;
use platform::domain::setting::UserSettingService;
use serde::Deserialize;
use std::sync::Arc;
use tauri::State;

/// 用户设置服务句柄（setup 时注册到 Tauri 状态）。
pub struct UserSettingHandle {
    pub store: Arc<InMemoryUserSettingStore>,
}

/// 单键读取请求。
#[derive(Debug, Deserialize)]
pub struct SettingGetRequest {
    pub owner_id: i64,
    pub key: String,
}

/// 单键写入请求。
#[derive(Debug, Deserialize)]
pub struct SettingSetRequest {
    pub owner_id: i64,
    pub key: String,
    pub value: String,
}

/// 读取单键设置。
#[tauri::command]
pub fn user_setting_get(
    state: State<'_, UserSettingHandle>,
    request: SettingGetRequest,
) -> common::DingDaResult<IpcResponse<Option<String>>> {
    let service = UserSettingService::new(state.store.as_ref());
    let result = service.get(request.owner_id, &request.key)?;
    Ok(IpcResponse::ok(result))
}

/// 写入单键设置。
#[tauri::command]
pub fn user_setting_set(
    state: State<'_, UserSettingHandle>,
    request: SettingSetRequest,
) -> common::DingDaResult<IpcResponse<()>> {
    let service = UserSettingService::new(state.store.as_ref());
    service.set(request.owner_id, &request.key, &request.value)?;
    Ok(IpcResponse::ok(()))
}
